SneakPeek/src/com/cyscorpions/dalejulian/sneakpeek/activities/EditSneakerEntryActivity.java

```diff
@@ -0,0 +1,178 @@
+		mBrand = (EditText) findViewById(R.id.txtSneakerBrand);
+		mRarity = (EditText) findViewById(R.id.txtSneakerRarity);
+		mSellVal = (EditText) findViewById(R.id.txtSellingValue);
+		mDesc = (EditText) findViewById(R.id.txtSneakerDescription);
+		mThumbnail = (ImageView) findViewById(R.id.imgSneakerEntry);
+	}
+
+	private void saveSneaker() {
+		SneakerCategory category = new SneakerCategory();
+		category.setName(mCategory.getText().toString());
+		Sneaker sneaker = new Sneaker(mName.getText().toString(), mBrand
+				.getText().toString(), mRarity.getText().toString(), mSellVal
+				.getText().toString(), mDesc.getText().toString(), category, 0);
+
+		SneakerDirectory.get(this).addSneaker(sneaker);
+		Intent intent = new Intent();
+		setResult(RESULT_OK, intent);
+		finish();
+	}
+
+	private void setupEditView() {
+		boolean isFromDetailActivity = getIntent().getBooleanExtra(
+				SneakerDetailsActivity.DETAILS_FROMDETAILS_EXTRA, false);
+		boolean isFromListActivity = getIntent().getBooleanExtra(
+				SneakerEntryListActivity.LIST_FROMLIST_EXTRA, false);
+
+		if (isFromDetailActivity) {
+			mSneaker.setBrand(getIntent().getStringExtra(
+					SneakerDetailsActivity.DETAILS_BRAND_EXTRA));
+			mSneaker.setName(getIntent().getStringExtra(
+					SneakerDetailsActivity.DETAILS_NAME_EXTRA));
+			mSneaker.setRarity(getIntent().getStringExtra(
+					SneakerDetailsActivity.DETAILS_RARITY_EXTRA));
+			mSneaker.setSellingValue(getIntent().getStringExtra(
+					SneakerDetailsActivity.DETAILS_SELLING_EXTRA));
+			mSneaker.setThumbnailId(getIntent().getIntExtra(
+					SneakerDetailsActivity.DETAILS_IMGID_EXTRA, 0));
+			mSneaker.setDescription(getIntent().getStringExtra(
+					SneakerDetailsActivity.DETAILS_DESC_EXTRA));
+			mSneaker.setCategory(getIntent().getStringExtra(
+					SneakerDetailsActivity.DETAILS_CATEGORY_EXTRA));
+			id = getIntent().getStringExtra(
+					SneakerDetailsActivity.DETAILS_UUID_EXTRA);
+		} else if (isFromListActivity) {
+			mSneaker.setBrand(getIntent().getStringExtra(
+					SneakerEntryListActivity.LIST_BRAND_EXTRA));
+			mSneaker.setName(getIntent().getStringExtra(
+					SneakerEntryListActivity.LIST_NAME_EXTRA));
+			mSneaker.setRarity(getIntent().getStringExtra(
+					SneakerEntryListActivity.LIST_RARITY_EXTRA));
+			mSneaker.setSellingValue(getIntent().getStringExtra(
+					SneakerEntryListActivity.LIST_SELLING_EXTRA));
+			mSneaker.setThumbnailId(getIntent().getIntExtra(
+					SneakerEntryListActivity.LIST_IMGID_EXTRA, 0));
+			mSneaker.setDescription(getIntent().getStringExtra(
+					SneakerEntryListActivity.LIST_DESC_EXTRA));
+			mSneaker.setCategory(getIntent().getStringExtra(
+					SneakerEntryListActivity.LIST_CATEGORY_EXTRA));
+			id = getIntent().getStringExtra(SneakerEntryListActivity.LIST_ID_EXTRA);
+		} else {
+			mCategory.setText(getIntent().getStringExtra(
+					SneakerEntryListActivity.LIST_CATEGORY_EDIT_EXTRA));
+			mSaveButton.setOnClickListener(new View.OnClickListener() {
```

Comment:
This clause is redundant. If ever this clause will run, the saveButton will again have a new click listener set on it on the next few statements